File: SpringBoot/src/main/java/com/project/foret/entity/Member.java
package com.project.foret.entity;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.util.*;

@Entity
@Getter
@Setter
@Table(name = "member")
public class Member {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String email;
    private String password;
    private String nickname;
    private String birth;
    private String deviceToken;
    @Temporal(TemporalType.TIMESTAMP)
    @CreationTimestamp
    private Date reg_date;

    @ManyToMany(targetEntity = Tag.class, cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    private List<Tag> tags;
    @ManyToMany(targetEntity = Region.class, cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    private List<Region> regions;
    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<MemberPhoto> photos;
    @OneToMany(mappedBy = "leader", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Foret> leadForets;
    @ManyToMany(targetEntity = Foret.class, mappedBy = "members", cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    private List<Foret> forets;
    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Board> boards;
    @OneToMany(mappedBy = "member", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Comment> comments;

    public void addTag(Tag tag) {
        if (tags == null) {
            tags = new ArrayList<>();
        }
        tags.add(tag);
    }

    public void addRegion(Region region) {
        if (regions == null) {
            regions = new ArrayList<>();
        }
        regions.add(region);
    }

    public void addPhoto(MemberPhoto memberPhoto) {
        if (photos == null) {
            photos = new ArrayList<>();
        }
        photos.add(memberPhoto);
        memberPhoto.setMember(this);
    }

    public void addForet(Foret foret) {
        if (forets == null) {
            forets = new ArrayList<>();
        }
        forets.add(foret);
    }

    public Member(Long id) {
        this.id = id;
    }

    public Member() {
    }

}

File: Android/app/src/main/java/com/project/foret/model/Member.kt
package com.project.foret.model

import java.util.*

data class Member(
    var id: Long?,
    val name: String?,
    val nickname: String?,
    val email: String?,
    val birth: String?,
    val password: String?,
    val reg_date: Date?,
    val tags: MutableList<Tag>?,
    val regions: MutableList<Region>?,
    val photos: MutableList<Photo>?
) {
    constructor(id: Long)
            : this(id, null, null, null, null, null, null, null, null, null)

    constructor(
        name: String,
        nickname: String,
        email: String,
        birth: String,
        password: String,
        tags: MutableList<Tag>,
        regions: MutableList<Region>,
    ) : this(null, name, nickname, email, birth, password, null, tags, regions, null)

    constructor(
        email:String,
        password: String
    ) : this(null, null, null, email, null, password, null, null,null,null)
}
